discovery/discovery.go

```diff
@@ -109,12 +110,15 @@ func (o *orchestratorPool) GetOrchestrators(ctx context.Context, numOrchestrator
 		}
 		return caps.CompatibleWith(info.Capabilities)
 	}
+	var wg sync.WaitGroup
```

Comment:
I think that instead of having this WaitGroup here, you can just move the following line inside `getOrchInfo()`:
```go
ctx, cancel := context.WithTimeout(clog.Clone(context.Background(), ctx), maxGetOrchestratorCutoffTimeout)
```

Reply:
nice good idea 👍